qt/aqt/mediasrv.py

```diff
@@ -28,146 +31,127 @@ def _getExportFolder():
-        if path.startswith(targetPath):
-            newPath = os.path.join(_exportFolder, path[len(targetPath) :])
-            return newPath
+    try:
+        if devMode:
+            print("Sending file '%s - %s'" % (directory, path))
+        return flask.send_from_directory(directory, path)
+
+    except Exception as error:
+        if devMode:
+            print(
+                "Caught HTTP server exception,\n%s"
+                % "".join(traceback.format_exception(*sys.exc_info())),
+            )
+
+        # swallow it - user likely surfed away from
+        # review screen before an image had finished
+        # downloading
+        return flask.Response(
+            "For path '%s - %s' %s!" % (directory, path, error),
+            status=HTTPStatus.INTERNAL_SERVER_ERROR,
+            mimetype="text/plain",
+        )
 
-        # catch /_addons references and rewrite them to addons folder
-        targetPath = os.path.join(os.getcwd(), "_addons", "")
-        if path.startswith(targetPath):
-            try:
-                addMgr = self.mw.addonManager
-            except AttributeError:
-                return path
 
-            addonPath = path[len(targetPath) :]
+def _redirectWebExports(path):
+    # catch /_anki references and rewrite them to web export folder
+    targetPath = "_anki/"
+    if path.startswith(targetPath):
+        return _exportFolder, path[len(targetPath) :]
 
-            try:
-                addon, subPath = addonPath.split(os.path.sep, 1)
-            except ValueError:
-                return path
-            if not addon:
-                return path
+    # catch /_addons references and rewrite them to addons folder
+    targetPath = "_addons/"
+    if path.startswith(targetPath):
+        addonPath = path[len(targetPath) :]
 
-            pattern = addMgr.getWebExports(addon)
-            if not pattern:
-                return path
+        try:
+            addMgr = _redirectWebExports.mw.addonManager
+        except AttributeError as error:
+            if devMode:
+                print("_redirectWebExports: %s" % error)
+            return _exportFolder, addonPath
 
-            subPath2 = subPath.replace(os.sep, "/")
-            if re.fullmatch(pattern, subPath) or re.fullmatch(pattern, subPath2):
-                newPath = os.path.join(addMgr.addonsFolder(), addonPath)
-                return newPath
+        try:
+            addon, subPath = addonPath.split(os.path.sep, 1)
+        except ValueError:
+            return addMgr.addonsFolder(), path
+        if not addon:
+            return addMgr.addonsFolder(), path
 
-        return path
+        pattern = addMgr.getWebExports(addon)
+        if not pattern:
+            return addMgr.addonsFolder(), path
 
+        if re.fullmatch(pattern, subPath):
+            return addMgr.addonsFolder(), addonPath
 
-# work around Windows machines with incorrect mime type
-RequestHandler.extensions_map[".css"] = "text/css"
```

Comment:
Does flask use a different mime type map? If it's using the same system settings then it's likely to suffer from the same issue.

Reply:
I just checked with `QTWEBENGINE_REMOTE_DEBUGGING=8080` and the response has the correct type:

![image](https://user-images.githubusercontent.com/5332158/84927744-a5709c80-b0a3-11ea-87db-2f040e7bca1f.png)

Reply:
The issue is for machines where the configured mime type is wrong:

https://anki.tenderapp.com/discussions/beta-testing/725-anki-210-beta-13-14-15